### DISCUSSION TITLE
Heatmap: Support array fields

### DISCUSSION THREAD

**Discussion by jetpax** [grafana/grafana#38253] [Dashboards, Panels & Visualizations]
I have an issue whilst trying to display FFT results in a heatmap.

I have a real[512] array of frequency bins and would like to plot a heatmap (or waterfall) of FFT bins against time.

For simplicity, assuming I had only 10 bins, eg

```
CREATE TABLE testarray (
time        TIMESTAMPTZ       NOT NULL,
values      real[10]
)
```

```

INSERT INTO testarray(time, values) VALUES(NOW(), ARRAY[0.1,1.2,2.3,3.4,4.5,5.6,6.7,7.8,8.9,9.0]);

```
Then in Grafana Heatmap using bucketed data  I can  use a query like

```
SELECT 
  "timestamp" AS "time",
  values[0] as "0",
  values[1] as "1",
  values[2] as "2",
  values[3] as "3",
  values[4] as "4",
  values[5] as "5",
  values[6] as "6",
  values[7] as "7",
  values[8] as "8",
  values[9] as "9",

FROM testarray
WHERE
   "timestamp" BETWEEN $__timeFrom() AND $__timeTo()
ORDER BY 1

```
 
and I I get the desired result of 10 FFT bins plotted in technicolor against time.

However, this is very cumbersome with 512 or 1024 bins, and if I try and pass **_values_** as an array, eg

```
SELECT 
  "timestamp" AS "time",
  values as "1"

FROM testarray
WHERE
   "timestamp" BETWEEN $__timeFrom() AND $__timeTo()
ORDER BY 1

```

although the SQL query returns a **_values_** column which is a real[],  Grafana responds with "No Data Points"

I would expect Grafana to interpret the real array as **_values_**, and use the array size to extrapolate the labels  from 1 thru' 10 (in this test case)

I am using TimescaleDB, but as I say, the standard query used above returns good data.

Am I just doing something wrong?

**Comment by torkelo** [comment]
each bucket needs it's own field, array fields are not supported currently

**Comment by jetpax** [comment]
Thanks for looking at that, could you give me a hint as to where I might add that in the source?
ie where is the serieslist generated?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
